server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/delivery/MailboxAppender.java

```diff
@@ -0,0 +1,122 @@
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.mailets.delivery;
+
+import java.util.Date;
+
+import javax.mail.Flags;
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+import org.apache.james.core.MimeMessageInputStream;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.exception.BadCredentialsException;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.slf4j.Logger;
+
+import com.google.common.base.Strings;
+
+public class MailboxAppender {
+    private static final boolean IS_RECENT = true;
+    private static final Flags FLAGS = null;
+
+    private final MailboxManager mailboxManager;
+    private final Logger logger;
+
+    public MailboxAppender(MailboxManager mailboxManager, Logger logger) {
+        this.mailboxManager = mailboxManager;
+        this.logger = logger;
+    }
+
+    public void append(MimeMessage mail, String user, String folder) throws MessagingException {
+        MailboxSession session = createMailboxSession(user);
+        append(mail, user, folder, session);
+    }
+
+    public void appendAndUseSlashAsSeparator(MimeMessage mail, String user, String folder, String fallbackFolder) throws MessagingException {
+        MailboxSession session = createMailboxSession(user);
+        append(mail, user, useSlashAsSeparator(folder, session, fallbackFolder));
+    }
+
+    private String useSlashAsSeparator(String urlPath, MailboxSession session, String fallbackFolder) {
+        String destination = urlPath.replace('/', session.getPathDelimiter());
+        if (Strings.isNullOrEmpty(destination)) {
+            destination = fallbackFolder;
+        }
+        if (destination.charAt(0) == session.getPathDelimiter()) {
+            destination = destination.substring(1);
+        }
+        return destination;
+    }
+
+    private void append(MimeMessage mail, String user, String folder, MailboxSession session) throws MessagingException {
+        mailboxManager.startProcessingRequest(session);
+        try {
+            MailboxPath mailboxPath = new MailboxPath(session.getPersonalSpace(), user, folder);
+            appendMessageToMailbox(mail, session, mailboxPath);
+        } catch (MailboxException e) {
+            throw new MessagingException("Unable to access mailbox.", e);
+        } finally {
+            closeProcessing(session);
+        }
+    }
+
+    private void appendMessageToMailbox(MimeMessage mail, MailboxSession session, MailboxPath path) throws MailboxException, MessagingException {
+        createMailboxIfNotExist(session, path);
```

Comment:
Well, that was the current behaviour.

First, I was told not to change the expected behaviour of mailet.
Second, the creation of INBOX outside of JMAP is done here. If I remove this line, we end up with no more delivered mails as INBOX can not be found nor created.

Maybe you want to add this mailbox creation behaviour inside the manager ? Or maybe we can add auto provisionning for the mailbox upon user creation ? However in both cases, it is, in my opinion, out of the scope of this ticket.